src/internal/hooks/use-base-component/index.ts

```diff
@@ -26,8 +28,14 @@ export default function useBaseComponent<T = any>(
   config?: ComponentConfiguration,
   analyticsMetadata?: AnalyticsMetadata
 ) {
-  useTelemetry(componentName, config);
+  const isVisualRefresh = useVisualRefresh();
+  const theme = getVisualTheme(THEME, isVisualRefresh);
+  useComponentMetrics(componentName, { packageSource: PACKAGE_SOURCE, packageVersion: PACKAGE_VERSION, theme }, config);
   useFocusVisible();
-  const elementRef = useComponentMetadata<T>(componentName, PACKAGE_VERSION, { ...analyticsMetadata });
+  const elementRef = useComponentMetadata<T>(
+    componentName,
+    { packageName: PACKAGE_SOURCE, version: PACKAGE_VERSION, theme },
+    analyticsMetadata as any
```

Comment:
This value is typed differently in the toolkit and this package

```
// this package
export interface AnalyticsMetadata {
  instanceIdentifier?: string;
  flowType?: FlowType;
  errorContext?: string;
  resourceType?: string;
}

// toolkit
export type AnalyticsMetadata = JSONObject;
```

This component's more strict version is not assignable to the toolkit. Before my change we used to fool typescript with runtime code `{ ...analyticsMetadata }`. I do not like it. If we mess with types anyway, let's do it more clearly at least